### PR TITLE
Adapt cross modal search to 2.0

### PR DESCRIPTION
**Pull request by Kelton8Z**

### CODE REVIEW

**Comment by Kelton8Z** [comment]
Primitive work in progress

File: cross-modal-search/flows/flow-index.yml [hunk 1]
```diff
@@ -34,9 +34,25 @@ pods:
     needs: [gateway]
   - name: text_indexer
     polling: any
-    uses: pods/index-text.yml
+    #uses: pods/index-text.yml (numpy + binary pb indexer)
```
Comment (theUnkownName) [inline cross-modal-search/flows/flow-index.yml:37]: Could you remove commented lines that are not necessary?

File: cross-modal-search/flows/flow-query.yml [hunk 2]
```diff
@@ -43,7 +43,7 @@ pods:
     shards: $JINA_SHARDS
     uses_after: pods/merge_matches_sort_topk.yml
     remove_uses_ba: true
-  - name: join_all
-    uses: _merge_root
-    needs: [text_indexer, image_kv_indexer]
-    read_only: true
+#  - name: join_all
```
Comment (theUnkownName) [inline cross-modal-search/flows/flow-query.yml:46]: Also delete those if we don't need them please

File: cross-modal-search/pods/index-image-kv.yml [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
-      - !KVIndexDriver
-        with:
-          executor: images_kv_idx
-    SearchRequest:
-      - !KVSearchDriver
-        with:
-          executor: images_kv_idx
-          traversal_paths: ['m']
-          recur_on: matches
+#!BinaryPbIndexer
```
Comment (theUnkownName) [inline cross-modal-search/pods/index-image-kv.yml:1]: We don't use any of this?

File: cross-modal-search/flows/flow-index.yml [hunk 4]
```diff
@@ -34,9 +34,25 @@ pods:
...
+    uses: NumpyIndexer
     shards: $JINA_SHARDS
+  - !NumpyIndexer
```
Comment (theUnkownName) [inline cross-modal-search/flows/flow-index.yml:40]: Can you add comments on the yml file of what is going on? As in the [other examples](https://github.com/jina-ai/examples/issues/445)

**Review by theUnkownName — CHANGES_REQUESTED** [review]
Just some details

File: cross-modal-search/flows/flow-query.yml (hunk 2 above)
Comment (Kelton8Z) [reply cross-modal-search/flows/flow-query.yml:46]: Deleted

File: cross-modal-search/flows/flow-index.yml (hunk 4 above)
Comment (Kelton8Z) [reply cross-modal-search/flows/flow-index.yml]: Sure, done

File: cross-modal-search/pods/index-image-kv.yml (hunk 3 above)
Comment (Kelton8Z) [reply cross-modal-search/pods/index-image-kv.yml:1]: The logic is adapted for 2.0 in executors.py

File: cross-modal-search/flows/flow-index.yml (hunk 1 above)
Comment (Kelton8Z) [reply cross-modal-search/flows/flow-index.yml]: Sure

**Comment by CatStark** [comment]
Hi @Kelton8Z , any news on this PR?

**Comment by Kelton8Z** [comment]
> Hi @Kelton8Z , any news on this PR?

The index flow is good, and query flow works with text input, working on incorporating image input

**Comment by nan-wang** [comment]
@Kelton8Z I've fixed the query part and the workspace.

**Review by nan-wang — APPROVED** [review]
LGTM👍

File: cross-modal-search/pods/executors.py [hunk 5]
```diff
@@ -229,30 +222,40 @@ def _get_sorted_top_k(
...
+    @requests(on='/search')
+    def query(self, docs: DocumentArray, **kwargs):
+        if not docs:
+            return
+        for doc in docs:
+            for match in doc.matches:
+                if match.id in self._docs:
+                    score = match.score
+                    match.MergeFrom(self._docs[match.id])
+                    match.score = score
```
Comment (JoanFM) [inline cross-modal-search/pods/executors.py:258]: why is this needed?
